sources/web/datalab/polymer/components/item-list/item-list.ts

```diff
@@ -251,12 +270,80 @@ class ItemListElement extends Polymer.Element {
       const shadow = '0px ' + yOffset + 'px 10px -5px #ccc';
       headerContainer.style.boxShadow = shadow;
     });
+
+    // Initial sort
+    this._sortBy(0);
   }
 
   resetFilter() {
     this._filterString = '';
   }
 
+  _formatColumnValue(value: ColumnTypeList, i: number, columns: Column[]): string {
+    if (columns[i]) {
+      if (columns[i].type === ColumnType.DATE) {
+        return (value as Date).toLocaleString();
+      } else {
+        return value.toString();
+      }
+    } else {
+      return '';
+    }
+  }
+
+  _columnButtonClicked(e: any) {
+    this._sortBy(e.model.itemsIndex);
+  }
+
+  _sortBy(column: number) {
+    // If sort is requested on the current sort column, reverse the sort order.
+    // Otherwise, set the current sort column to that.
+    if (this._currentSort.column === column) {
+      this._currentSort.asc = !this._currentSort.asc;
+    } else {
+      this._currentSort = {
+        asc: true,
+        column,
+      };
+    }
+
+    this.$.list.sort = (a: ItemListRow, b: ItemListRow) => {
+      // Bail out of sort if no columns have been set yet.
+      if (!this.columns.length) {
+        return;
+      }
+      let compResult = -1;
```

Comment:
Shouldn't you be returning 0 if the two items are equal? Otherwise, it seems possible you will have problems when sorting lists with equal elements.

Reply:
Is there any advantage to return 0 as opposed to -1 if two items are equal?

Reply:
The contract for a javascript comparator function is to return 0 if the two values are equal. Broken comparator functions can cause incorrect sorting. Even if it works today, the sort function might change in a future version in a way that reveals the broken comparator and causes the sort not to order correctly.

Reply:
Done.